lib/board.rb

```diff
@@ -0,0 +1,59 @@
+class Board
+
+  attr_accessor :cells
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def initialize
+    reset!
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def full?
+    @cells.all? do |position|
+      position != " "
+    end
+  end
+
+  def turn_count
+    counter = 0
+    @cells.each do |player|
+      if player == "X" || player == "O"
+        counter +=1
+      end
+    end
+    counter
+  end
+
+  def taken?(input)
+    if @cells[input.to_i - 1] ==  " "
+      false
+    else
+      true
+    end
+  end
+
+  def valid_move?(input)
+    !taken?(input) && input.to_i > 0 && input.to_i < 10
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`input.to_i.between?(1,9)`